src/components/MTableEditField/CurrencyField.js

```diff
@@ -24,7 +23,8 @@ function CurrencyField({ forwardedRef, ...props }) {
         }
       }}
       inputProps={{
-        'aria-label': props.columnDef.title
+        'aria-label': props.columnDef.title,
+        style: { textAlign: 'right' }
```

Comment:
Is there a way not to use inline style? I know that there are still a lot of legacy uses of it, but we are trying to reduce those to improve performance?

Reply:
It would need to generate a class to apply. I think performance in this case would be a wash.

Inputs default as left-aligned. Please advise if you find a better way.

Reply:
It could lead to death by a thousand cuts, since it affects each cell. Especially with large table the mounting time could be up to 1.5x slower and the dom size up to 2x of a size. That's why would like to avoid it. Also the maintainability is a lot better as well.
I would probably use the @material-ui/styles package. IT is fast, exposes the useStyles hook or withStyles HOC and has no increased bundle size since we are already using /core.

Reply:
That was what I was referring to with generating styles. Their example snippet:
```
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';

const useStyles = makeStyles({
  root: {
    background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)',
    border: 0,
    borderRadius: 3,
    boxShadow: '0 3px 5px 2px rgba(255, 105, 135, .3)',
    color: 'white',
    height: 48,
    padding: '0 30px',
  },
});

export default function Hook() {
  const classes = useStyles();
  return <Button className={classes.root}>Hook</Button>;
}
```

Every time that function re-renders, it calls `useStyles()` for every cell.  There is some performance impact from that as well. That's why I suggest this approach should be at a project-wide level instead of in each sub-component.

Currently the project is using the old `m-table-edit-field.js` module which doesn't use hooks. The code that you're referring to is part of the refactor which hasn't been activated yet. I suggest we evaluate performance at that time so we don't fall into the trap of premature optimization.

Reply:
Performance is no worse than before. The inline style had been applied to the parent TextField, which has now been moved into its underlying input.

I like that you're thinking about performance. However this is verging on scope creep because this ticket doesn't change performance, and its intent is specific to functionality. It would be better to open a separate ticket to investigate performance and benchmark theories before investing precious resources.

Reply:
Yes you are right. This looks good. Will merge.